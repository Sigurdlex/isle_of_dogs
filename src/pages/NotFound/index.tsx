import { push } from 'connected-react-router';
import React, { useCallback } from 'react';
import { connect } from 'react-redux';
import { Dispatch } from 'redux';
import styled from 'styled-components';

import { Header, ArrowButton } from 'components';
import { ORANGE } from 'constants/colors';

interface NotFoundProps {
  dispatch: Dispatch;
}

export const NotFound = ({ dispatch }: NotFoundProps) => {
  const handleGoToMain = useCallback(() => dispatch(push('/')), []);
  return (
    <>
      <Header>
        <ArrowButton
          isLeft
          handleClick={handleGoToMain}
          color={ORANGE}
        >
          Back to list
        </ArrowButton>
      </Header>
      <Container color={ORANGE}>
        <ErrorH1>404 - page not found</ErrorH1>
      </Container>
    </>
  );
};

export default connect()(NotFound);

const Container = styled.div<{ color: string }>`
  background: ${({ color }) => color};
  padding-top: 100px;
  height: 100vh;
  display: flex;
  min-width: 700px;
`;

const ErrorH1 = styled.h1`
  text-transform: uppercase;
  text-align: center;
  margin: auto;
  background: white;
  border-radius: 4px;
  box-shadow: 5px 5px 10px rgba(0, 0, 0, .5);
  font-family: "Franklin Gothic Medium", "Franklin Gothic", "ITC Franklin Gothic", Arial, sans-serif;
  font-size: 3rem;
  padding: 0.5em;
`;
