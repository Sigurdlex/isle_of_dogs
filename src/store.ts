import { routerMiddleware } from 'connected-react-router';
import { createBrowserHistory } from 'history';
import { createStore, applyMiddleware, Store } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { composeWithDevTools } from 'redux-devtools-extension';

import { reducers } from 'reducers';
import { rootSaga } from 'sagas';

export const history = createBrowserHistory();

export const configureStore = (initialState = {}): Store => {
  const sagaMiddleware = createSagaMiddleware();
  const enhancers = [applyMiddleware(sagaMiddleware, routerMiddleware(history))];
  const store = createStore(
    reducers(history),
    initialState,
    composeWithDevTools(...enhancers)
  );
  sagaMiddleware.run(rootSaga);
  return store;
};
