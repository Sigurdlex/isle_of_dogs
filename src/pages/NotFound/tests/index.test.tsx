import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';

import { configureStore } from 'store';
import NotFound from '..';

test('<NotFound />', () => {
  const store = configureStore();
  const {
    container: { firstChild: container },
  } = render(
    <Provider store={store}>
      <NotFound />
    </Provider>,
  );
  expect(container).toMatchSnapshot();
});
