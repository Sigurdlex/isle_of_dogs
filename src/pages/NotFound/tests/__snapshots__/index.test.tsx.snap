// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> 1`] = `
<header
  class="sc-htpNat etzdtK"
>
  <div
    class="sc-bxivhb hzQWTh"
  >
    <button
      class="sc-bdVaJa iJFbTX"
      color="#ffbd03"
    >
      Back to list
    </button>
  </div>
</header>
`;
